src/System.IO.FileSystem/tests/DirectoryInfo/Refresh.cs

```diff
@@ -1,126 +1,48 @@
-            if (Directory.Exists(newDirName))
-                Directory.Delete(newDirName);
-
-            iCountTestcases++;
-            Directory.CreateDirectory(dirName);
-            dir2 = new DirectoryInfo(dirName);
-            dir2.MoveTo(newDirName);
-            dir2.Refresh();
-            Directory.Delete(newDirName);
-            //----------------------------------------------------------------
-
-            // [] Change Attributes and refresh
-            //----------------------------------------------------------------
-            strLoc = "Loc_00006";
-            iCountTestcases++;
-
-            Directory.CreateDirectory(dirName);
-            dir2 = new DirectoryInfo(dirName);
-
-            if (((Int32)dir2.Attributes & (Int32)FileAttributes.ReadOnly) != 0)
-            {
-                Console.WriteLine(dir2.Attributes);
-                iCountErrors++;
-                printerr("Error_00007! Attribute set before refresh");
-            }
-            dir2.Attributes = FileAttributes.ReadOnly;
-            dir2.Refresh();
-            iCountTestcases++;
-            if (((Int32)dir2.Attributes & (Int32)FileAttributes.ReadOnly) <= 0)
-            {
-                iCountErrors++;
-                printerr("Error_00008! Object not refreshed after setting readonly");
-            }
-
-            dir2.Attributes = new FileAttributes();
-            dir2.Refresh();
-            if (((Int32)dir2.Attributes & (Int32)FileAttributes.ReadOnly) != 0)
-            {
-                iCountErrors++;
-                printerr("Error_00009! Object not refreshed after removing readonly");
-            }
-
-            //----------------------------------------------------------------
-
-            if (Directory.Exists(dirName))
-                Directory.Delete(dirName);
-
-            ///////////////////////////////////////////////////////////////////
-            /////////////////////////// END TESTS /////////////////////////////
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            testDir.Delete();
+            testDir.Refresh();
+            Assert.False(testDir.Exists);
         }
-        catch (Exception exc_general)
+
+        [Fact]
+        public void NameChange()
         {
-            ++iCountErrors;
-            Console.WriteLine("Error Err_8888yyy!  strLoc==" + strLoc + ", exc_general==" + exc_general.ToString());
+            //NOTE: MoveTo adds a trailing slash to the path of the DirectoryInfo
+            string source = GetTestFilePath();
+            string dest = GetTestFilePath();
+            DirectoryInfo testDir = Directory.CreateDirectory(source);
+            testDir.MoveTo(dest);
+            testDir.Refresh();
+            Assert.Equal(testDir.FullName, dest + Path.DirectorySeparatorChar);
         }
-        ////  Finish Diagnostics
-        if (iCountErrors != 0)
+
+        [Fact]
+        public void AttributeChange()
```

Comment:
The closest thing is the tests in "set_attributes", although those don't call Refresh() or create a new FileAttribute(), they just set and check and move on to the next attribute.